File: src/main/java/moa/moabackend/domain/user/AuthController.java
package moa.moabackend.domain.user;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import moa.moabackend.global.security.TokenResponse;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/signup")
    public ResponseEntity<Void> signup(@Valid @RequestBody UserDto.SignupRequest request) {
        authService.signup(request);
        return ResponseEntity.ok().build();
    }

    @PostMapping("/login")
    public ResponseEntity<TokenResponse> login(@Valid @RequestBody UserDto.LoginRequest request) {
        return ResponseEntity.ok(authService.login(request));
    }

    @PostMapping("/reissue")
    public ResponseEntity<TokenResponse> reissue(@Valid @RequestBody UserDto.ReissueRequest request) {
        return ResponseEntity.ok(authService.reissue(request));
    }
}

File: src/main/kotlin/moa/moabackend/domain/auth/presentation/AuthController.kt
package moa.moabackend.domain.auth.presentation

import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import moa.moabackend.domain.auth.presentation.dto.request.AdminSignupRequest
import moa.moabackend.domain.auth.presentation.dto.request.LoginRequest
import moa.moabackend.domain.auth.presentation.dto.request.ReissueRequest
import moa.moabackend.domain.auth.presentation.dto.request.SignupRequest
import moa.moabackend.domain.auth.presentation.dto.response.TokenResponse
import moa.moabackend.domain.auth.service.AdminSignupService
import moa.moabackend.domain.auth.service.LoginService
import moa.moabackend.domain.auth.service.ReissueService
import moa.moabackend.domain.auth.service.SignupService

@RestController
@RequestMapping("/auth")
@Tag(name = "인증 API", description = "회원가입, 로그인, 토큰 재발급을 담당합니다.")
class AuthController(
    private val signupService: SignupService,
    private val adminSignupService: AdminSignupService,
    private val loginService: LoginService,
    private val reissueService: ReissueService
) {

    @Operation(summary = "일반 회원가입")
    @PostMapping("/signup")
    @ResponseStatus(HttpStatus.CREATED)
    fun signup(@RequestBody @Valid request: SignupRequest): TokenResponse {
        return signupService.execute(request)
    }

    @Operation(summary = "관리자(판매자) 회원가입")
    @PostMapping("/signup/admin")
    @ResponseStatus(HttpStatus.CREATED)
    fun adminSignup(@RequestBody @Valid request: AdminSignupRequest): TokenResponse {
        return adminSignupService.execute(request)
    }

    @Operation(summary = "로그인")
    @PostMapping("/login")
    fun login(@RequestBody @Valid request: LoginRequest): TokenResponse {
        return loginService.execute(request)
    }

    @Operation(summary = "토큰 재발급")
    @PostMapping("/reissue")
    fun reissue(@RequestBody @Valid request: ReissueRequest): TokenResponse {
        return reissueService.execute(request)
    }
}
